hoomd/md/Thermostat.h

```diff
@@ -337,38 +337,128 @@ class MTTKThermostat : public Thermostat
 class BussiThermostat : public Thermostat
     {
     public:
+    /** Construct the thermostat.
+
+        @param T Temperature set point over time.
+        @param group Group of particles this thermostat is applied to.
+        @param thermo Use to compute the thermodynamic properties of the group.
+        @param sysdef Used to access the simulation seed and MPI communicator.
+        @param tau Thermostat time constant.
+    */
     BussiThermostat(std::shared_ptr<Variant> T,
                     std::shared_ptr<ParticleGroup> group,
                     std::shared_ptr<ComputeThermo> thermo,
-                    std::shared_ptr<SystemDefinition> sysdef)
-        : Thermostat(T, group, thermo, sysdef)
+                    std::shared_ptr<SystemDefinition> sysdef,
+                    Scalar tau)
+        : Thermostat(T, group, thermo, sysdef), m_tau(tau)
         {
         }
 
     std::array<Scalar, 2> getRescalingFactorsOne(uint64_t timestep, Scalar deltaT) override
         {
         m_thermo->compute(timestep);
 
-        const auto ntdof = m_thermo->getTranslationalDOF();
-        const auto nrdof = m_thermo->getRotationalDOF();
+        Scalar e_factor;
+        Scalar rt_left;
+        Scalar rr_left;
+        const int ntdof = static_cast<int>(m_thermo->getTranslationalDOF());
+        const int nrdof = static_cast<int>(m_thermo->getRotationalDOF());
```

Comment:
```suggestion
        // Integration methods will always operate on subsets with integer degrees of freedom.
        // Partial degrees of freedom are removed by getTranslationalDOF() when the subset of
        // particles is not the same subet selected for integration.
        const int ntdof = static_cast<int>(m_thermo->getTranslationalDOF());
        const int nrdof = static_cast<int>(m_thermo->getRotationalDOF());
```
It isn't immediately obvious that the number of degrees of freedom will always be an integer here. Add a comment to clarify.
https://github.com/glotzerlab/hoomd-blue/blob/4184da26c976722d9852100c14941b62a018ca05/hoomd/md/IntegratorTwoStep.cc#L181-L186

Reply:
Does it even matter if ndof is an integer or not? I see an edge case if 0 < ndof < 1 and m_tau != 0, but otherwise I think it works out fine; mathematically speaking I mean, I'm not sure why and how someone would end up with non-integer dofs.